File: solutions/dart/sieve/1/lib/sieve.dart
// In sacred honor of Shree DR.MDD 🌹

class Sieve {
  final int limit;
  final List<int> _primeList = [];
  Sieve(this.limit);

  List<int> get primes {
    if (limit >= 2 && _primeList.isEmpty) {
      var sieve = List.generate(limit + 1, (i) => i);
      sieve[1] = 0;

      void eliminate(int base, {int gap = 2}) {
        for (var k = base * base; k <= limit; k += gap) {
          sieve[k] = 0;
        }
      }

      eliminate(2);
      for (int candidate = 3; candidate * candidate <= limit; candidate += 2) {
        if (sieve[candidate] != 0) {
          eliminate(candidate, gap: 2 * candidate);
        }
      }

      _primeList.addAll(sieve.where((val) => val != 0));
    }
    return _primeList;
  }
}

File: solutions/fortran/sieve/1/sieve.f90
! For my Shree DR.MDD  
module sieve
   implicit none

   private

   public primes

contains

   function primes(limit) result(nums)
      integer, intent(in) :: limit
      integer, allocatable :: nums(:)
      integer :: idx, k, pick

      if (limit < 2) then
         allocate(nums(0))
         return
      end if

      nums = [(idx, idx=2, limit)]

      idx = 1
      do while (idx <= size(nums))
         pick = nums(idx)
         nums = pack(nums, [(.true., k=1, idx), mod(nums((idx+1):), pick) /= 0])
         idx = idx + 1
      end do

   end function primes

end module sieve
